core-web/libs/block-editor/src/lib/elements/dot-bubble-menu/components/dot-link-editor-popover/dot-link-editor-popover.component.ts

```diff
@@ -0,0 +1,294 @@
+import { Observable, Subject } from 'rxjs';
+
+import { HttpClient } from '@angular/common/http';
+import {
+    Component,
+    inject,
+    input,
+    signal,
+    OnDestroy,
+    ViewChild,
+    ElementRef,
+    computed,
+    HostListener
+} from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+import { AutoCompleteModule } from 'primeng/autocomplete';
+import { ButtonModule } from 'primeng/button';
+import { InputTextModule } from 'primeng/inputtext';
+import { Listbox, ListboxModule } from 'primeng/listbox';
+import { SkeletonModule } from 'primeng/skeleton';
+
+import { debounceTime, distinctUntilChanged, takeUntil, pluck } from 'rxjs/operators';
+
+import { Editor } from '@tiptap/core';
+
+import { DotCMSContentlet } from '@dotcms/dotcms-models';
+
+import { EditorModalDirective } from '../../../../directive/editor-modal.directive';
+
+interface SearchResultItem {
+    displayName: string;
+    url: string;
+    hasTitleImage?: boolean;
+    inode?: string;
+}
+
+/**
+ * A popover component for creating and editing links in the DotCMS block editor.
+ * This component provides functionality to:
+ * - Search for internal content to link to
+ * - Create links to external URLs
+ * - Edit existing link properties (URL, target attribute)
+ * - Remove links from selected text or images
+ */
+@Component({
+    selector: 'dot-link-editor-popover',
+    templateUrl: './dot-link-editor-popover.component.html',
+    styleUrls: ['./dot-link-editor-popover.component.scss'],
+    standalone: true,
+    imports: [
+        FormsModule,
+        ListboxModule,
+        AutoCompleteModule,
+        InputTextModule,
+        SkeletonModule,
+        ButtonModule,
+        EditorModalDirective
+    ]
+})
+export class DotLinkEditorPopoverComponent implements OnDestroy {
+    @ViewChild('popover', { read: EditorModalDirective }) popover: EditorModalDirective;
+    @ViewChild('input', { read: ElementRef }) searchInput?: ElementRef<HTMLInputElement>;
+    @ViewChild('resultListbox') searchResultsListbox?: Listbox;
+
+    protected readonly editor = input.required<Editor>();
+    protected readonly searchQuery = signal<string>('');
+    protected readonly isSearching = signal<boolean>(false);
+    protected readonly searchResults = signal<SearchResultItem[]>([]);
+
+    // Current link state for editing existing links
+    protected readonly existingLinkUrl = signal<string | null>(null);
+    protected readonly linkTargetAttribute = signal<string>('_blank');
+
+    protected readonly isExternalURL = computed(() => {
+        const validURLRegex = /^(https?:\/\/)?([\da-z.-]+)\.([a-z.]{2,6})([/\w .-]*)*\/?$/;
```

Comment:
do you need a regex or can it be sufficient to try to parse it to URL using the URL API?

Reply:
Changed!